pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1335,6 +1350,62 @@ public void deleteRealtimeTable(String tableName) {
   }
 
   /**
+   * Build the partition mapping table that maps a tuple of (partition number, replica group number) to a list of
+   * servers. Two important configurations are explained below.
+   *
+   * - 'numInstancesPerPartition': this number decides the number of servers within a replica group.
+   *
+   * - 'partitionColumn': this configuration decides whether to use the table or partition level replica groups.
+   *
+   * @param tableName: Name of table
+   * @param tableConfig: Configuration for table
+   * @return Partition mapping table from the given configuration
+   */
+  private PartitionToReplicaGroupMappingZKMetadata buildPartitionToReplicaGroupMapping(String tableName,
```

Comment:
We don't need 'mirrorAssignmentAcrossReplicaGroups' when creating the mapping table. The mapping table specifies the replica group for each partition or table and 'mirrorAssignmentAcrossReplicaGroups' affects when picking a server from the replica group. In other words, its setting does not change any information in the mapping table side. We definitely need it for the segment assignment part.